AEPAnalytics/Tests/FunctionalTests/AnalyticsAppExtConsequenceTests.swift

```diff
@@ -0,0 +1,82 @@
+/*
+ Copyright 2022 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import XCTest
+import AEPServices
+@testable import AEPAnalytics
+@testable import AEPCore
+
+class AnalyticsAppExtConsequenceTests: AnalyticsFunctionalTestBase {
+    override func setUp() {
+        super.setupBase(forApp: false)
+        dispatchDefaultConfigAndIdentityStates()
+    }
+    
+
+    func testHandleAnalyticsConsequence() {
+        MobileCore.setLogLevel(.trace)
+        let eventData: [String: Any] = [
+            AnalyticsConstants.EventDataKeys.TRIGGERED_CONSEQUENCE: [
+                AnalyticsConstants.EventDataKeys.ID: "id",
+                AnalyticsConstants.EventDataKeys.TYPE: "an",
+                AnalyticsConstants.EventDataKeys.DETAIL : [
+                    "action" : "testActionName",
+                    "contextdata": ["k1" : "v1" , "k2" : "v2"]
+                ]
+            ]
+        ]
+        let ruleEngineEvent = Event(name: "Rule event", type: EventType.rulesEngine, source: EventSource.responseContent, data: eventData)
+        mockRuntime.simulateComingEvent(event: ruleEngineEvent)
+        waitForProcessing()
+        
+        let expectedVars = [
+            "ce": "UTF-8",
+            "pev2" : "AMACTION:testActionName",
```

Comment:
@kevinlind This is an example of the difference in the tests. Anywhere where we check for CP variable in the `Analytics` functional tests, we no longer have that variable from the AppExtension child class.

Reply:
As the test classes are very similar but with a few changes, it would be great to find a way to reuse the majority of the code. I'm not sure how to do that, however. Though, it is something that can be taken up in another PR.

Reply:
This is a good point. I can get this worked out in this PR or it might never get done.